src/py/flwr/cli/run/run.py

```diff
@@ -102,10 +102,17 @@ def on_channel_state_change(channel_connectivity: str) -> None:
         """Log channel connectivity."""
         log(DEBUG, channel_connectivity)
 
+    if root_certificates := federation.get("root-certificates"):
+        root_certificates_bytes = Path(root_certificates).read_bytes()
+        insecure = False
+    else:
+        root_certificates_bytes = None
+        insecure = True
```

Comment:
In both cases, we need to read an `insecure` flag from the `federation` config.

If `root_certificates` are provided, it's ok if the user omits it. If the user sets it, it must be set to `false`.

If `root_certificates` are not provided, then the user must provide it and it must be set to true.

Reply:
Btw @danieljanes , should those settings be under `[tool.flwr.federations.name]` or under `[tool.flwr.federations.name.options]`?